src/transformers/models/auto/modeling_tf_auto.py

```diff
@@ -553,7 +553,7 @@ def __init__(self):
 
     @classmethod
     @replace_list_option_in_docstrings(TF_MODEL_MAPPING, use_model_types=False)
-    def from_config(cls, config):
+    def from_config(cls, config, **kwargs):
```

Comment:
We can change it here to `name: str = None` instead - the kwargs are just used to pass the correct name of the model in case a composite model is instantiated.

Reply:
We do need to pass the `name` argument though because composite model have to make use of `TFAutoModel.from_config(...)` at init

Reply:
Actually, the only other way to make it work would be to introduce two if-else statements, which I don't think is cleaner..since it's the auto model class, I think it's alright to do some under-the-hood argument passing